src/state/zustand/swapperStore/actions.ts

```diff
@@ -29,12 +32,16 @@ export const updateTradeAmounts =
     )
 
 export const toggleIsExactAllowance =
-  (set: SetSwapperStoreAction<SwapperStore>): SwapperAction['toggleIsExactAllowance'] =>
+  (
+    set: SetSwapperStoreAction<SwapperStore>,
+    getState: () => SwapperState,
+  ): SwapperAction['toggleIsExactAllowance'] =>
   () =>
     set(
-      state => {
-        state.isExactAllowance = !state.isExactAllowance
-        return state
+      draft => {
+        const currentState = getState()
```

Comment:
Any reason to pass immer's `getState` and not use the draft state, which, before being mutated, should effectively be the same?